packages/react-components/react-utilities/src/hooks/useControllableState.ts

```diff
@@ -42,40 +38,21 @@ function isFactoryDispatch<State>(newState: React.SetStateAction<State>): newSta
 export const useControllableState = <State>(
   options: UseControllableStateOptions<State>,
 ): [State, React.Dispatch<React.SetStateAction<State>>] => {
-  const isControlled = useIsControlled(options.state);
   const initialState = typeof options.defaultState === 'undefined' ? options.initialState : options.defaultState;
   const [internalState, setInternalState] = React.useState<State>(initialState);
```

Comment:
One more improvement might to move the check to factory, so it will be executed once:

```ts
const [internalState, setInternalState] = React.useState<State>(
  () => typeof options.defaultState === 'undefined' ? options.initialState : options.defaultState
);
```